packages/amplify-e2e-tests/src/cleanup-e2e-resources.ts

```diff
@@ -519,7 +564,12 @@ const mergeResourcesByCCIJob = (
 };
 
 const deleteAmplifyApps = async (account: AWSAccountInfo, accountIndex: number, apps: AmplifyAppInfo[]): Promise<void> => {
-  await Promise.all(apps.map(app => deleteAmplifyApp(account, accountIndex, app)));
+  if(apps.length > DELETE_LIMITS.PER_BATCH.OTHER){
```

Comment:
These if statements aren't needed. Just call apps.slice(0, 50) and if the array has < 50 elements, it will return the whole array